### PR TITLE
Issue 378

### LINKED ISSUE #378: Add API endpoint to get session state
## Related Issue

- #373

## Description

State endpoint provides session state.
Supported states:
- `pending`
- `running`
- `completed`
- `failed` 

## Endpoint

- GET `/SASjsApi/session/{sessionId}/state`

## URL Params

| Param | Type | Required | Description |
| -------- | ------- | ------- | ------- |
| `sessionId` | string | true | Session ID |

## Responses

| Status | Meaning | Description | Payload |
| -------- | ------- | ------- | ------- |
| `200` | [OK](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) | Request has succeeded | Session State |
| `404` | [Not Found](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) | Session with requested ID was not found | Error Message |

## Response Example

| Status | Payload |
| -------- | ------- |
| `200` | running |
| `404` | {"message": "Session with ID '20241030142120-34994-17302980803870' was not found."} |

### PR DESCRIPTION
**Pull request by YuryShkoda**
## Issue

Closes #378

## Intent

- Add `/SASjsApi/session/{sessionId}/state` API endpoint.

## Implementation

- Replaced multiple properties of the `Session `interface indicating session state with a single `state` property at `api/src/types/Session.ts`.
- Refactored session-related code to apply updated `Session` interface.
- Added GET `/{sessionId}/state` route to session router at `api/src/routes/api/session.ts`.
- Added `sessionIdValidation` at `api/src/utils/validation.ts`.
- Added `sessionState` method to `SessionController` at `api/src/controllers/session.ts`.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [ ] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
- [x] All CI checks are green.
- [x] Reviewer is assigned.

### CODE REVIEW

File: api/src/types/Session.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
+export enum SessionState {
+  initialising = 'initialising', // session is initialising and nor ready to be used yet
```
Comment (sabhas) [inline api/src/types/Session.ts:2]: small typo in comment, "nor" should be "not"

File: api/src/controllers/internal/processProgram.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export const processProgram = async (
     await moveFile(codePath + '.bkp', codePath)
 
     // we now need to poll the session status
-    while (!session.completed) {
+    while (session.state === SessionState.completed) {
```
Comment (sabhas) [inline api/src/controllers/internal/processProgram.ts:52]: Can you check this condition again? I think it should be `session.state !== SessionState.completed`

**Review by sabhas — CHANGES_REQUESTED** [review]

File: api/src/controllers/internal/processProgram.ts (hunk 2 above)
Comment (YuryShkoda) [reply api/src/controllers/internal/processProgram.ts:52]: you are right it should be `!==`

**Review by medjedovicm4gl — APPROVED** [review]
Don't have anything else to add

**Pull request merged**